### ISSUE TITLE
v0.5.2 detector: SilentCipher not detected on AAC-in-MP4/M4A; same audio detects fine when ffmpeg-decoded to PCM

### ISSUE THREAD

**Issue by neitzert** [CreativeMayhemLtd/provcheck#24]
## Summary

The mark from v0.5.2 `provcheck-kit watermark` **survives an AAC encode round-trip** (0.961 → 0.935 confidence). But the **detector silently returns "not_detected" when reading AAC audio from inside MP4 (or M4A) containers** — i.e. exactly the format any delivery pipeline would produce. When the same MP4's audio is fully re-decoded to PCM by ffmpeg and handed to the detector as a .wav, the mark is detected normally.

So the issue isn't on the embed side, isn't on the codec, and isn't on the detector's PCM math. It's on the detector's **AAC bitstream / MP4 audio-packet decode** path.

## Versions
- `provcheck-kit 0.5.2`
- `provcheck 0.5.2` (the detector binary)

## Environment
- Linux container (Ubuntu 24.04.1 LTS, Docker Desktop / WSL2)
- glibc 2.39
- x86_64
- `libpcsclite1` runtime dep present

## Repro chain

Starting from a 27.74s stereo 48 kHz MP3 (a music bumper). Verified detection at each step:

```sh
# 1. Embed with v0.5.2 defaults (--channels auto, --sdr-db 30, verify-after-embed ON).
$ provcheck-kit watermark long.mp3 -o long_marked.wav --overwrite
provcheck-kit:   1222920 samples (27.73 s @ 44100 Hz, source 2 channels, output 2 channels)
provcheck-kit:   embedding [44, 46, 4d, 01, 00] (SDR 30 dB)
provcheck-kit:   verify-after-embed: silentcipher conf 0.961 OK (payload recovered)
provcheck-kit: done.

# 2. Detector on the WAV — works.
$ provcheck --json long_marked.wav | jq '.watermarks[0]'
  silent_cipher: detected   conf=0.961   payload=[68, 70, 77, 1, 0]   ✓

# 3. Standard delivery: ffmpeg → MP4 (h264 + aac stereo 44.1k 192k).
$ ffmpeg -y -loop 1 -framerate 25 -i brand.png -i long_marked.wav \
    -vf "scale=1920:1080,format=yuv420p" -c:v libx264 -crf 23 \
    -c:a aac -b:a 192k -ar 44100 -ac 2 -shortest final_long.mp4

# 4. Detector on MP4 — FAILS.
$ provcheck --json final_long.mp4 | jq '.watermarks[0]'
  silent_cipher: not_detected   conf=0.000   ❌

# 5. Stream-copy the audio out of the MP4 into M4A (no decode):
$ ffmpeg -y -i final_long.mp4 -vn -c:a copy extracted.m4a
$ provcheck --json extracted.m4a | jq '.watermarks[0]'
  silent_cipher: not_detected   conf=0.000   ❌ (same bug, no MP4 video around it)

# 6. Now have ffmpeg FULLY DECODE the same AAC stream to PCM and detect that:
$ ffmpeg -y -i final_long.mp4 -vn -c:a pcm_s16le extracted.wav
$ provcheck --json extracted.wav | jq '.watermarks[0]'
  silent_cipher: detected   conf=0.935   payload=[68, 70, 77, 1, 0]   ✓

# 7. As an extra control, encode the marked WAV to MP3 (no MP4 around it):
$ ffmpeg -y -i long_marked.wav -c:a libmp3lame -b:a 320k long.mp3
$ provcheck --json long.mp3 | jq '.watermarks[0]'
  silent_cipher: detected   conf=0.959   ✓
```

## What this proves

- **Embed is fine.** v0.5.2's verify-after-embed and the standalone WAV detect both confirm the mark at 0.961.
- **AAC psycho-acoustic encode is fine.** The mark survives at 0.935 inside the AAC bitstream — provable because once you ffmpeg-decode that AAC back to PCM, the detector finds the mark.
- **The container is the problem.** Stream-copying the AAC out of `.mp4` into `.m4a` doesn't change the audio bytes; both yield not_detected. So it's not a per-container bug — it's that the **detector's symphonia-based AAC reader from MP4-style containers isn't producing the same PCM** that ffmpeg's decoder produces from the same bitstream.

Possible root causes (in rough probability order):
1. Sample-rate or channel-layout metadata in the MP4 audio header is being read differently than ffmpeg reads it, so the detector's STFT operates on misaligned/resampled data.
2. AAC packet framing (LATM vs raw vs ADTS-in-MP4) — the kit's decoder may be assuming raw AAC frames and producing slightly-shifted PCM samples that destroys phase alignment for the watermark math.
3. PCM is being passed to detector as `i16` after a different normalization than what ffmpeg uses, so amplitude scales differently.

The smoking gun is that the **EXACT SAME AAC bitstream** detects fine when ffmpeg decodes it and detects 0.000 when provcheck decodes it.

## Suggested workaround until fixed

In the detector code, when the input file is an MP4/M4A container, shell out to ffmpeg for the AAC → PCM step (e.g. `ffmpeg -i input -vn -f s16le -ar 44100 - | detector_core`) rather than doing the decode in symphonia. Lossless data path, no behavior difference.

## Why this matters for delivery pipelines

Every audio bumper / podcast / video compilation gets shipped as MP4 (or MP3 in a podcast feed). For the watermark to be **operationally useful for end-to-end provenance**, the detector must read the same MP4 the audience downloads. Right now any third-party verifier running `provcheck file.mp4` against a doomscroll-style MP4 will see `not_detected` even though the mark is present in the PCM at 0.935 confidence.

Closing #23 once this lands would close the production use-case end-to-end.

**Comment by neitzert** [comment]
**v0.5.3 shipped:** https://github.com/CreativeMayhemLtd/provcheck/releases/tag/v0.5.3

You called this exactly right: the embed always survived AAC; the detector was the broken half. Two changes in v0.5.3:

1. `crates/provcheck-{watermark,audioseal,wavmark}/src/audio.rs` now hardcodes `AAC_DEFAULT_PRIMING_SAMPLES = 1024` when symphonia returns `delay = None` for an AAC track. symphonia 0.5.5's `isomp4` reader does not expose the `edts/elst` edit list or the `iTunSMPB` tag as `codec_params.delay`, so prior to this fix every STFT frame was 1024 samples (one AAC LC frame) out of phase with the embedder's frame grid. Confirmed via a `decode_probe` diff against ffmpeg on the same bitstream: ours yielded 662528 samples, ffmpeg 661504, diff exactly 1024.

2. `mp4`, `m4b`, and `mov` are now in the audio-extension allowlist of all three detector crates. Without (2), MP4 video containers with an AAC audio track were rejected by the early `looks_like_audio` sniff before symphonia ever saw them, which is why `extracted.m4a` failed but `final_long.mp4` failed differently in your repro.

Post-fix numbers on a representative 15s clip, SilentCipher SDR 30, ffmpeg aac 192k stereo:

- pristine WAV:                conf 0.959
- AAC m4a (was 0.0):           conf 0.921
- AAC mp4 h264+aac (was 0.0):  conf 0.921

Side effect: this invalidates the "AAC delivery unsupported for silentcipher" line in `docs/v0.5.2-codec-survival/`. That doc has been amended in place with a "v0.5.3 correction" section. AudioSeal at alpha=3.0 still has the higher post-AAC margin (conf 0.999 vs 0.92), but silentcipher is now also a viable AAC delivery path — useful for pipelines that prefer the 40-bit ASCII payload over AudioSeal's 16-bit ECC brand ID.

I also added `scripts/check-before-push.sh` plus a `.git/hooks/pre-push` wrapper so the cargo test workspace + parity-vs-upstream sweep + AAC delivery survival now runs as a mandatory pre-push gate on the dev repo. The smoke would have caught this on the v0.5.2 push if it had been wired then. Wall-clock about 10 minutes; bypass is `git push --no-verify` with the reason noted in the commit body.

Closing this. If you re-run the issue repro on v0.5.3 and any cell still shows 0.000 conf, reopen and I'll dig in.
